homeassistant/components/homewizard/switch.py

```diff
@@ -0,0 +1,147 @@
+"""Creates Homewizard Energy switch entities."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    DEVICE_CLASS_SWITCH,
+    SwitchEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import COORDINATOR, DOMAIN, DeviceResponseEntry
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switches."""
+
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
```

Comment:
A good change, for a follow up PR or here, could be to add a class that describes the ` hass.data[DOMAIN]` like tuya does.

https://github.com/home-assistant/core/blob/1a5716efc4cff95fc6837935c404fae524279f7b/homeassistant/components/tuya/__init__.py#L44-L49

Than you don't have to type it here and use a string import to define the `coordinator` key.

Reply:
Good idea.
As this is a bit out of scope for this PR and it also has to be implemented in `sensor.py`, I think it is better to make this a separate PR.

Reply:
I think we can remove this abstraction at full, as there is nothing else added to here (it just adds another unneeded dictionary at this point?)

```py
coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
```

Reply:
Like this? https://github.com/home-assistant/core/pull/64084/commits/47d9e4e8727bcd0d9aaabafd004d5d6b7402def7

Reply:
I think Frenck meant that you create a dictionary with a single key ("coordinator") and thus don't need this.

You could just store it in your hass data directly, without first creating a dictionary.
`hass.data[DOMAIN][entry.entry_id] = coordinator`

Reply:
I see, nice! Completely overlooked this, it was a dict as some other values were stored here before but it should have been removed indeed.

Updated it here (994001b). But it also changes `sensor.py` which is a bit out of scope of this PR. If desirable, I can roll back the change and open a separate PR.